File: src/fabaindaiz/modulator/core/main/modulator.java
package fabaindaiz.modulator.core.main;

import fabaindaiz.modulator.Modulator;
import fabaindaiz.modulator.modules.AModule;

public class modulator extends AModule {
    private final Modulator plugin;

    public modulator(Modulator modulator) {
        this.plugin = modulator;
        setLang(plugin.getConfiguration().getMainLang());
    }

    @Override
    public void onEnable() {
        plugin.getCommand("modulator").setExecutor(new modulatorCommand(plugin, this));
        plugin.getCommand("modulator").setTabCompleter(new modulatorTabCompleter(plugin, this));
        plugin.getCommand("modulator").setPermissionMessage(getLang().get("error.noper"));
    }

    @Override
    public void onDisable() {

    }
}

File: src/main/kotlin/cl/fabaindaiz/modulator/Modulator.kt
package cl.fabaindaiz.modulator

import cl.fabaindaiz.modulator.core.command.ModulatorCommand
import org.bukkit.Bukkit
import org.bukkit.plugin.java.JavaPlugin

class Modulator: JavaPlugin() {
    companion object {
        var instance: Modulator? = null
        private set;
    }

    override fun onEnable() {
        val command =  ModulatorCommand()

        instance = this
        Bukkit.getLogger().info("Enabled!")
    }

    override fun onDisable() {
    }


}